operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1796 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package lbipam
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/cilium/ipam/service/ipallocator"
+	"github.com/sirupsen/logrus"
+	"go.uber.org/multierr"
+	"golang.org/x/exp/slices"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+
+	operatorOption "github.com/cilium/cilium/operator/option"
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s"
+	cilium_api_v2alpha1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	cilium_client_v2alpha1 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_core_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	slim_meta_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	client_typed_v1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/typed/core/v1"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/stream"
+)
+
+const (
+	// The condition added to services to indicate if a request for IPs could be satisfied or not
+	ciliumSvcRequestSatisfiedCondition = "io.cilium/lb-ipam-request-satisfied"
+
+	ciliumPoolIPsTotalCondition     = "io.cilium/ips-total"
+	ciliumPoolIPsAvailableCondition = "io.cilium/ips-available"
+	ciliumPoolIPsUsedCondition      = "io.cilium/ips-used"
+	ciliumPoolConflict              = "io.cilium/conflict"
+
+	// The annotation LB IPAM will look for when searching for requested IPs
+	ciliumSvcLBIPSAnnotation = "io.cilium/lb-ipam-ips"
+
+	// The string used in the FieldManager field on update options
+	ciliumFieldManager = "cilium-operator-lb-ipam"
+
+	serviceNamespaceLabel = "io.kubernetes.service.namespace"
+	serviceNameLabel      = "io.kubernetes.service.name"
+)
+
+var Cell = cell.Module(
+	"LB-IPAM",
+	// Provide LBIPAM so instances of it can be used while testing
+	cell.Provide(newLBIPAM),
+	// Invoke an empty function which takes an LBIPAM to force its construction.
+	cell.Invoke(func(*LBIPAM) {}),
+)
+
+type LBIPAMParams struct {
+	cell.In
+
+	Logger logrus.FieldLogger
+
+	LC         hive.Lifecycle
+	Shutdowner hive.Shutdowner
+
+	ClientSet    k8sClient.Clientset
```

Comment:
```suggestion
	Clientset    k8sClient.Clientset
```

Let's stick with the original form: https://pkg.go.dev/k8s.io/client-go/kubernetes#Clientset.